### ISSUE TITLE
/count + sort returns empty set

### ISSUE THREAD

**Issue by smirea** [florianholzapfel/express-restify-mongoose#284]
If you do a query to `/User?sort=-_id` say it returns 100 results. If you do a query to `/User/count?sort=-_id` it returns a single empty object result `{}`

I know count + sort does not make much sense, but I think the behavior should be to ignore the `sort` operation when doing a count as opposed to silently failing (or at the very least throwing an error)

**Comment by Zertz** [comment]
This should indeed throw an error:

```
db.collection.count().sort({_id: 1})
//= TypeError: db.collection.count(...).sort is not a function
```

I just realized I didn't publish it, but I think that same fix should allow `.sort()` to throw on `.count()` queries. We just need to add a test case to confirm.

**Comment by Zertz** [comment]
@smirea Hmm, seems like Mongoose ignores sort internally on count queries. The latest commit correctly returns `{ count: 3 }`

Do you have a `count` field in private or protected?

**Comment by smirea** [comment]
nope, straight up setup

**Comment by Zertz** [comment]
Can you try with this:

```js
dependencies: {
  "express-restify-mongoose": "florianholzapfel/express-restify-mongoose#369b29b"
}
```

**Comment by Zertz** [comment]
This should be fixed in 4.1.0
